### PR TITLE
chore: upgrades most dev dependencies

### PR DESCRIPTION
**Pull request by priyajeet**
Bunch of files have been re-prettified.
Some locale strings had spaces { likeThis }, change to {likeThis}.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: .eslintrc.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ module.exports = {
         'react/no-access-state-in-setstate': 'off', // fixme
         'react/no-array-index-key': 'off', // fixme
         'react/no-this-in-sfc': 'off',
+        'formatjs/enforce-description': 'error' // fixme
```
Comment (priyajeet) [inline .eslintrc.js]: TODO: move to frontend repo

File: babel.config.js [hunk 2]
```diff
@@ -18,22 +18,13 @@ module.exports = {
...
         ],
     ],
     env: {
         development: {
-            plugins: [
-                'flow-react-proptypes',
-                [
-                    'react-intl',
-                    {
-                        enforceDescriptions: false,
```
Comment (priyajeet) [inline babel.config.js]: Part of eslint now, removed from react-intl

File: src/elements/content-sidebar/activity-feed/task-new/messages.js [hunk 3]
```diff
@@ -76,22 +76,22 @@ const messages = defineMessages({
...
     },
     tasksFeedHeadlineApproval: {
         id: 'be.contentSidebar.activityFeed.taskNew.tasksFeedHeadlineApproval',
-        defaultMessage: '{ user } assigned an Approval Task',
+        defaultMessage: '{user} assigned an Approval Task',
         description: 'Comment headline for an approval task',
     },
     tasksFeedHeadlineGeneralCurrentUser: {
         id: 'be.contentSidebar.activityFeed.taskNew.tasksFeedHeadlineGeneralCurrentUser',
-        defaultMessage: '{ user } assigned you a Task',
```
Comment (priyajeet) [inline src/elements/content-sidebar/activity-feed/task-new/messages.js]: react-intl plugin converted { foo } to {foo} but it doesn't anymore.

File: .eslintrc.js [hunk 4]
```diff
@@ -17,7 +17,9 @@ module.exports = {
...
     },
+    plugins: ['formatjs'],
```
Comment (jstoffan) [inline .eslintrc.js]: Is this something new that we're doing in addition to prettier? Should it be a separate PR from the dependencies upgrade or is it needed due to changes in babel and/or react-intl?

File: src/api/__tests__/File.test.js [hunk 5]
```diff
@@ -528,7 +528,11 @@ describe('api/File', () => {
         });
         test('should do nothing if destroyed', () => {
             file.isDestroyed = jest.fn().mockReturnValue(true);
-            file.getFileExtension('id', () => {}, () => {});
+            file.getFileExtension(
+                'id',
```
Comment (jstoffan) [inline src/api/__tests__/File.test.js]: Is the prettier upgrade causing these code style changes?

File: .eslintrc.js (hunk 4 above)
Comment (priyajeet) [reply .eslintrc.js]: I removed it, doesn't work well for us right now

File: src/api/__tests__/File.test.js (hunk 5 above)
Comment (priyajeet) [reply src/api/__tests__/File.test.js]: yup

File: src/api/uploads/__tests__/MultiputPart.test.js [hunk 6]
```diff
@@ -183,7 +183,10 @@ describe('api/uploads/MultiputPart', () => {
 
         test.each`
             parts
-            ${[{ offset: 1, part_id: 1 }, { offset: 1, part_id: 1 }]}
+            ${[
+    { offset: 1, part_id: 1 },
+    { offset: 1, part_id: 1 },
+]}
```
[the remark this answers is not in the document]
Comment (priyajeet) [reply src/api/uploads/__tests__/MultiputPart.test.js]: 🐳

File: babel.config.js (hunk 2 above)
Comment (priyajeet) [reply babel.config.js]: reverted

File: .eslintrc.js (hunk 1 above)
Comment (priyajeet) [reply .eslintrc.js]: reverted for now

File: .eslintrc.js (hunk 4 above)
Comment (priyajeet) [reply .eslintrc.js]: @jstoffan I added this back, they fixed that above issue. There is 1 remaining issue I filed https://github.com/formatjs/formatjs/issues/304 which only effects 1 of our files.

In any case, I will do a followup PR to remove this plugin from here and move to frontend repo.
BTW, yeah its needed with the new babel react intl plugin which is also now upgraded

File: babel.config.js (hunk 2 above)
Comment (priyajeet) [reply babel.config.js]: removed again

File: src/components/i18n/FormattedCompMessage.js [hunk 7]
```diff
@@ -159,14 +159,12 @@ class FormattedCompMessage extends React.Component<Props, State> {
...
         // react-intl will do the correct plurals if necessary
-        const translation = intl.formatMessage(
-            {
-                id,
-                defaultMessage: source,
-                description,
-            },
-            values,
-        );
+        const descriptor = {
```
Comment (priyajeet) [inline src/components/i18n/FormattedCompMessage.js]: https://github.com/formatjs/formatjs/issues/304

File: src/features/left-sidebar/LeftSidebar.js [hunk 8]
```diff
@@ -185,7 +185,7 @@ class LeftSidebar extends React.Component<Props, State> {
 
     getIcon(
         iconElement?: ?React.Element<any>,
-        IconComponent?: ?React.ComponentType<any>,
+        IconComponent?: ?React.ComponentType<any>, // eslint-disable-line
```
Comment (priyajeet) [inline src/features/left-sidebar/LeftSidebar.js]: not sure why this was giving a problem with new eslint, but this was the only occurrence
